### ISSUE TITLE
Exported class name within namespace not validated

### ISSUE THREAD

**Issue by deckerj** [Microsoft/tslint-microsoft-contrib#105]
The `exported-name` rule does not work correctly when the exported member (e.g. of a class) is wrapped in a namespace block (aka [internal module](https://github.com/Microsoft/TypeScript-Handbook/blob/master/pages/Namespaces.md))

The following declaration within a file `File.ts` should be detected and tslint sould issue a warning:
   
    namespace com.example {
       class ThisIsNotTheNameOfTheFile {
       }
     }

**Comment by HamletDRC** [comment]
thanks!

**Comment by HamletDRC** [comment]
I assume that your class declaration should have the 'export' keyword applied, correct?

**Comment by HamletDRC** [comment]
I pushed this into the releases branch so you can test it on your project. 

Just point your npm dependency to: 

        "tslint-microsoft-contrib": "git+https://github.com/Microsoft/tslint-microsoft-contrib.git#5bab6616b3be1c77fd30683e5368a4834c31bb4b",

**Comment by HamletDRC** [comment]
We do not use namespaces in codebase, so it would be really helpful if you ran this at least once against your own codebase and reported any false positives or issues.

**Comment by deckerj** [comment]
Awesome - thanks a lot for the quick fix. I will check out your changes next week.

**Comment by deckerj** [comment]
One more question regarding namespaces. Would it be complicated to check if the namespaces match the directory structure the files are located?

    File Foo.ts:
    namespace aaa.bbb.ccc {
       export class Foo { }
    }

-> Rule : Foo.ts must have path aaa/bbb/ccc/Foo.ts

I know that this is probably not the usual way, TypeScript codebases are structured. But coming from the Java-World, this approach fits well to the package declarations in Java.

Would it be complicated to write such a rule? How would you do it? Is there a similar rule that checks the pathname against statements?

**Comment by HamletDRC** [comment]
@deckerj It would be possible to write a rule like this, but I cannot see why you would actually want your code to be written this way. Your example code would be so difficult to use: 

    import aaa = require('aaa/bbb/ccc/Foo');
    let fooInstance: aaa.bbb.ccc.Foo = new aaa.bbb.ccc.Foo(); 

I don't think you want that.

**Comment by deckerj** [comment]
My use case is targeted to internal typescript module usage rather than external modules:

```
File Foo.ts:
namespace aaa.bbb.foo {
      export class Foo { }
}

File Bar.ts:
namespace aaa.bbb.bar {
   
   import Foo = aaa.bbb.foo.Foo;
   
   export class Bar { 
      let foor = new Foo();
   }

}
```

**Comment by HamletDRC** [comment]
It should be possible. Do you think it should be part of this rule (as an optional parameter)? I think probably yes.

**Comment by deckerj** [comment]
The rule works very well on our code base. Thanks again!

Touch, I figured out that the rule does not work with enums:

    namespace com.example {
       export enum ThisIsNotTheNameOfTheFile {
       }
    }
